device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/amqps/AmqpsIotHubConnection.java

```diff
@@ -235,6 +237,10 @@ public void open(Queue<DeviceClientConfig> deviceClientConfigs, ScheduledExecuto
                 {
                     closeConnectionWithException("Timed out waiting for worker links to open", true);
                 }
+
+                // In case this is a recovery from a prematurely closed connection, we will re-queue subscribe messages that were sent within this connection.
+                requeueSubscribeMessages();
```

Comment:
Does this resend the device method subscription as well?

Reply:
no, only Device twin at this point.

Reply:
from what I understand device methods do not require a message to be sent to subscribe to properties, they just need the link to be opened. but if in the future they require messages to be sent, we will ad them to the subscribe list and send those along with twin subscriptions

Reply:
Right, that's true!